Two lines same name in the same production

julnor124 opened the pull request:
Now you can’t create lines with the same name in a single production.

- normalizes names (trim + lowercase) for case-insensitive match
- memoized duplicate check against a default line
- live validation + error on duplicate
- disables "Add Line" / "Create Production" when a duplicate is detected
- useEffect revalidates on defaultLine or lines change

Three different cases where you could add a new line:
1. Standard
<img width="591" height="538" alt="Skärmavbild 2025-10-31 kl  14 55 16" src="https://github.com/user-attachments/assets/a5d8d006-034b-4534-a533-bf0978d1cd7b" />

2. When editing a name:
<img width="464" height="357" alt="Skärmavbild 2025-10-31 kl  15 00 17" src="https://github.com/user-attachments/assets/3a147de9-6a06-42ad-95f4-5533d3a281d1" />

3. When adding a new line:
<img width="467" height="493" alt="Skärmavbild 2025-10-31 kl  15 01 07" src="https://github.com/user-attachments/assets/6eb41e3b-bbd6-4003-a992-a83e865d9a65" />

And here you can see that you can still have the same line name in different productions:
<img width="939" height="231" alt="Skärmavbild 2025-10-31 kl  15 07 32" src="https://github.com/user-attachments/assets/a6164af9-4ca0-4e56-a76c-1d9307682afc" />

Saelmala reviewed (commented):
Overall LGTM, one improvement that can be done to reduce duplicate code is to create e.g. a useHasDuplicateLineName hook that can be used in both create-production-page.tsx and manage-production-buttons.tsx. 

For example, something like this:
```
export const useHasDuplicateLineName = <TLine extends { name?: string }>({
    candidateName,
    lines,
    getLineName = defaultGetLineName,
  }: UseHasDuplicateLineNameProps<TLine>) => {
    return useMemo(() => {
      const normalizedCandidate = normalizeName(candidateName);
      if (!normalizedCandidate || !lines?.length) return false;

      return lines.some(
        (line) => normalizeName(getLineName(line)) === normalizedCandidate
      );
    }, [candidateName, lines, getLineName]);
  };
```
then from the components, it can just be called with e.g.:
```
 const hasDuplicateWithExistingLines = useHasDuplicateLineName({
    candidateName: lineName,
    lines: production.lines,
  });
```
```
  const hasDuplicateWithDefaultLine = useHasDuplicateLineName({
    candidateName: defaultLine,
    lines,
  });
```

julnor124 commented:
Added useHasDuplicateLineName hook and shared normalizeLineName helper to handle duplicate-line checks. Refactored CreateProductionPage and ManageProductionButtons to use them instead of local logic.
